internal/database/repositories/asset_version_repository.go

```diff
@@ -45,6 +47,16 @@ func (a *assetVersionRepository) Read(assetVersionName string, assetID uuid.UUID
 	return asset, err
 }
 
+func (a *assetVersionRepository) Delete(tx core.DB, assetVersion *models.AssetVersion) error {
+	err := a.db.Delete(&assetVersion).Error //Call db delete function with the provided asset version
```

Comment:
This looks like double reference:

```suggestion
	err := a.db.Delete(assetVersion).Error //Call db delete function with the provided asset version
```